R/step-mutate.R

```diff
@@ -19,18 +19,26 @@ dt_call.dtplyr_step_mutate <- function(x, needs_copy = x$needs_copy) {
   if (!x$nested) {
     j <- call2(":=", !!!x$new_vars)
   } else {
-    assign <- Map(function(x, y) call2("<-", x, y), syms(names(x$new_vars)), x$new_vars)
-    new_vars <- unique(names(x$new_vars))
-    output <- call2(".", !!!syms(new_vars))
-    expr <- call2("{", !!!assign, output)
-    j <- call2(":=", call2("c", !!!new_vars), expr)
+    mutate_list <- mutate_nested_vars(x$new_vars)
+    j <- call2(":=", call2("c", !!!mutate_list$new_vars), mutate_list$expr)
   }
 
   out <- call2("[", dt_call(x$parent, needs_copy), , j)
 
   add_grouping_param(out, x, arrange = FALSE)
 }
 
+mutate_nested_vars <- function(mutate_vars) {
+  assign <- Map(function(x, y) call2("<-", x, y), syms(names(mutate_vars)), mutate_vars)
```

Comment:
Can use `map2()` from #255

Reply:
I merged with #255 and then found out it's actually from #252 ;-)